packages/ui-components/src/components/text-field/text-field.tsx

```diff
@@ -126,6 +131,9 @@ export class KvTextField {
 										}}
 									/>
 								)}
+								<div class={{ 'right-slot-container': true, 'focus': this.focused }}>
```

Comment:
Only to understand the tech decision... Why we decide to use a slot here to draw anything (like KvIcon)?

Reply:
@ricardogoncalves89

Reply:
@miguelpinto98 , the reason to use the slot was taken in order to be able to re-use the textfield with other components like the dropdown. With this solution, the logic behind opening the dropdown or clicking the reset button can stay on the Search/Dropdown component

Reply:
Not a huge fan of the approach. Since the `right-slot` will always be an icon, we could pass down only the icon name and emit an event when the icon is clicked, which would be then handled by the parent. The end result is the same and less error prone.

With that being said, I see the point of having a fully customizable right slot in which his behaviour is not handled by the text-field component as it should not. 

For that sake, I'll not oppose this approach, just wanted to share with you an alternative way 👍